sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -458,6 +461,24 @@ case class RowToColumnarExec(child: SparkPlan) extends RowToColumnarTransition {
     // This avoids calling `schema` in the RDD closure, so that we don't need to include the entire
     // plan (this) in the closure.
     val localSchema = this.schema
+    if (enableArrowColumnVector) {
+      val maxRecordsPerBatch = SQLConf.get.arrowMaxRecordsPerBatch
+      val timeZoneId = SQLConf.get.sessionLocalTimeZone
+      return child.execute().mapPartitionsInternal { rowIterator =>
+        val context = TaskContext.get()
+        ArrowConverters.toColumnarBatchIterator(
+          ArrowConverters
+            .toArrowRecordBatchIterator(
+              rowIterator,
+              localSchema,
+              maxRecordsPerBatch,
+              timeZoneId,
+              context),
+          localSchema,
+          timeZoneId,
+          context)
+      }
+    }
```

Comment:
@BryanCutler and @HyukjinKwon , for the implementation of RowToColumnarExec, I now use `toArrowRecordBatchIterator` + `toColumnarBatchIterator` to do the conversion, so there will be no serialization and deserialization.